allocator/nonmoving.go

```diff
@@ -1,6 +1,12 @@
 package allocator
 
-import "github.com/tetratelabs/wazero/experimental"
+import (
+	"errors"
+
+	"github.com/tetratelabs/wazero/experimental"
+)
+
+var errOutOfMemory = errors.New("allocator: requested size exceeds reserved address space")
```

Comment:
This confused me a bit…
This is being used for something that wazero _never_ asks us to do, right? (realloc to size > max)
If so, I think a more appropriate name would be `errInvalidRequest` or something.

Reply:
Makes sense, reworded it